front/components/assistant/conversation/input_bar/editor/extensions/MentionPasteExtension.tsx

```diff
@@ -0,0 +1,123 @@
+import { Plugin, PluginKey } from "@tiptap/pm/state";
+import { Extension } from "@tiptap/react";
+
+import { createMarkdownSerializer } from "@app/components/assistant/conversation/input_bar/editor/markdownSerializer";
+import type { WorkspaceType } from "@app/types";
+
+interface MentionPasteExtensionOptions {
+  owner: WorkspaceType;
+}
+
+const mentionPastePluginKey = new PluginKey("mentionPaste");
+
+/**
+ * Parse mentions from backend.
+ */
+async function parseMentionsOnBackend(
+  text: string,
+  ownerSId: string
+): Promise<string> {
+  const response = await fetch(`/api/w/${ownerSId}/assistant/mentions/parse`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ markdown: text }),
+  });
+
+  if (!response.ok) {
+    throw new Error("Failed to parse mentions");
+  }
+
+  const data = await response.json();
+  return data.markdown;
+}
+
+/**
+ * Extension to handle pasting with mentions.
+ * This extension works AFTER TipTap processes the paste event.
+ * It intercepts the pasted content, converts it to markdown, sends it to the backend
+ * to parse mentions, and then replaces the content with properly parsed mentions.
+ *
+ * Flow:
+ * 1. User pastes content (HTML or text)
+ * 2. TipTap processes and converts to its internal format
+ * 3. This extension intercepts and converts to markdown
+ * 4. Backend parses @mentions and returns markdown with serialized mentions
+ * 5. Extension converts back to TipTap nodes with mention nodes
+ */
+export const MentionPasteExtension =
+  Extension.create<MentionPasteExtensionOptions>({
+    name: "MentionPaste",
+
+    addOptions() {
+      return {
+        owner: {} as WorkspaceType,
+      };
+    },
+
+    addProseMirrorPlugins() {
+      const { owner } = this.options;
+      const editor = this.editor;
+
+      return [
+        new Plugin({
+          key: mentionPastePluginKey,
+          props: {
+            handlePaste: (view, event, slice) => {
+              // Get text from the slice after TipTap processing.
+              const text = slice.content.textBetween(
+                0,
+                slice.content.size,
+                "\n"
+              );
+
+              // Only process if text contains @.
+              if (!text.includes("@")) {
+                return false;
+              }
+
+              // Prevent default and handle manually.
+              event.preventDefault();
```

Comment:
nitpick and probably it doesn't change anything but this should come first inside the `handlePaste` function no?